src/components/HotspotListItem.tsx

```diff
@@ -25,69 +24,74 @@ const HotspotListItem = ({
   showCarot = false,
 }: HotspotListItemProps) => {
   const { t } = useTranslation()
-
+  const { toggleConvertHntToCurrency } = useCurrency()
+  const handlePress = useCallback(() => onPress?.(hotspot), [hotspot, onPress])
   return (
-    <TouchableOpacityBox
-      backgroundColor="grayBox"
-      flexDirection="row"
-      justifyContent="space-between"
-      alignItems="center"
-      padding="m"
-      borderRadius="m"
-      onPress={() => onPress?.(hotspot)}
-    >
-      <Box
+    <Box marginHorizontal="l" marginBottom="s">
+      <TouchableOpacityBox
+        backgroundColor="grayBox"
         flexDirection="row"
         justifyContent="space-between"
         alignItems="center"
-        flex={1}
+        padding="m"
+        borderRadius="m"
+        onPress={handlePress}
       >
-        <Box flexDirection="column">
-          <Box flexDirection="row" alignItems="center" marginBottom="xxs">
-            {hotspot.status?.online === 'online' ? (
-              <CheckCircle width={17} height={17} />
-            ) : (
-              <Attention width={17} height={17} />
-            )}
+        <Box
+          flexDirection="row"
+          justifyContent="space-between"
+          alignItems="center"
+          flex={1}
+        >
+          <Box flexDirection="column">
+            <Box flexDirection="row" alignItems="center" marginBottom="xxs">
+              {hotspot.status?.online === 'online' ? (
+                <CheckCircle width={17} height={17} />
+              ) : (
+                <Attention width={17} height={17} />
+              )}
+              <Text
+                variant="body2Medium"
+                color="offblack"
+                paddingStart="s"
+                ellipsizeMode="tail"
+                numberOfLines={2}
+                maxWidth={210}
+              >
+                {animalName(hotspot.address)}
+              </Text>
+            </Box>
+            <Box marginTop="xs">
+              <Text variant="body3Light" color="blueGray">
+                {hotspot.location
+                  ? `${hotspot.geocode?.longStreet}, ${hotspot.geocode?.longCity}, ${hotspot.geocode?.shortCountry}`
+                  : t('hotspot_details.no_location_title')}
+              </Text>
+            </Box>
             <Text
-              variant="body2Medium"
-              color="offblack"
-              paddingStart="s"
-              ellipsizeMode="tail"
-              numberOfLines={2}
-              maxWidth={210}
+              onPress={toggleConvertHntToCurrency}
```

Comment:
Can we add a debounce to this

Reply:
yeah, good call

Reply:
I guess you added the ability to tap from multiple places which is awesome. Maybe the debounce can be in a lower level so we dont have to add it everywhere

Reply:
yeah, I think I'll debounce the `toggleConvertHntToCurrency` function.